### PR TITLE
`General`: Fix statistics view showing incorrect scores

### LINKED ISSUE #5742: Missing course statistics depending on the previous section
### Describe the bug

https://user-images.githubusercontent.com/60005702/196202304-1d78dc31-4001-45f6-be8c-8983e2a9feda.mp4

### To Reproduce

1. Go to the Exercises
2. Switch directly to the Statistics
3. See missing Statistics
4. Reload the page
5. Statistics should be visible

### Expected behavior

---

### Screenshots

_No response_

### What browsers are you seeing the problem on?

Firefox

### Additional context

_No response_

### Relevant log output

_No response_

### LINKED ISSUE #6103: Statistics of a Course not displayed correctly
### Describe the bug

The statistics functionality for a course (as a student) is not working as expected.
Before clicking the 'Refresh' button on the Statistics page of a course, the displayed data differs from the data after clicking the 'Refresh' button.
Only the "after-refresh" data is actually correct.

### To Reproduce

1. Go to 'Course Overview'
2. Click on a specific course
3. Go to 'Statistics' in the selected course
4. Note down the displayed points (and percentage etc.)
5. Click on 'Refresh'
6. Note that the displayed points etc. changed upon refreshing

### Expected behavior

See the correct statistics, even without manually clicking the 'Refresh' button on the Statistics page of a Course.

### Screenshots

**Screenshot 1: Before clicking the 'Refresh' button**

image

**Screenshot 2: After clicking the 'Refresh' button**

image

### Which version of Artemis are you seeing the problem on?

5.12.9

### What browsers are you seeing the problem on?

Chrome

### Additional context

_No response_

### Relevant log output

_No response_

### LINKED ISSUE #6254: Statistics view shows incorrect points on first load
### Describe the bug

Besides having already one exercise completed, the statistics view still indicates that I received 0 points.
After reloading the page, the score is updated to the correct value.

### To Reproduce

See Screenrecording

### Expected behavior

The statistics view should display the correct values already on the first load.

### Screenshots

https://user-images.githubusercontent.com/9954674/219050983-b4f1f622-e428-464d-b13f-d2f717fd578f.mov

### Which version of Artemis are you seeing the problem on?

6.0.5

### What browsers are you seeing the problem on?

Safari

### Additional context

_No response_

### Relevant log output

_No response_

### LINKED ISSUE #6263: Incorrect visualization of the current exercise's status
### Describe the bug

On the student's exercise dashboard, the visualization of the current exercise's status differs from the status shown in the exercise's normal visualization.
This behavior occurs on the first load as well as on a reload. Switching the ribbon (for example to Lectures) fixes the visualization.

### To Reproduce

See screen recording

### Expected behavior

The exercise state should be consistent on the whole dashboard.

### Screenshots

https://user-images.githubusercontent.com/9954674/219441842-12c897ef-29ba-4c62-a3a4-4132039e9275.mov

### Which version of Artemis are you seeing the problem on?

6.0.5

### What browsers are you seeing the problem on?

Chrome

### Additional context

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by maximiliansoelch**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.

### Motivation and Context

Fixes #6254 
Fixes #5742 
Fixes #6103 
Fixes #6263

### Description

If the cached participations are empty, they overwrite the actual student participations received from the server. 
A check was added, that only overwrites the student participation in case the cached participation is not empty.

### Steps for Testing

Prerequisites:
- 1 Student
- 1 Programming Exercise with more than 0 points achieved

1. Log in to Artemis
2. Navigate to the Course Overview & click on a course
3. Click on Statistics
4. Make sure the overall points are displayed correctly (for reference compare it to the screen recording below)
5. Reload the page / Click the refresh button and make sure the scores stay the same

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

Before:

https://user-images.githubusercontent.com/9954674/219817460-4f95caa1-a3ab-4e43-bf5d-e4cb8205471d.mov

---
After:

https://user-images.githubusercontent.com/9954674/219817225-27b846d8-b4c1-4158-99b7-dfc4680488c5.mov

### CODE REVIEW

File: src/main/webapp/app/overview/course-exercises/course-exercise-row.component.ts [hunk 1]
```diff
@@ -78,7 +78,7 @@ export class CourseExerciseRowComponent implements OnInit, OnDestroy, OnChanges
 
     ngOnChanges(): void {
         const cachedParticipations = this.participationWebsocketService.getParticipationsForExercise(this.exercise.id!);
-        if (cachedParticipations) {
+        if (cachedParticipations?.length > 0) {
```
Comment (ge65cer) [inline src/main/webapp/app/overview/course-exercises/course-exercise-row.component.ts:81]: This should also work.

```suggestion
        if (cachedParticipations?.length) {
```

**Review by luisa1999 — APPROVED** [review]
Code looks fine to me!

**Review by PhilipWerz — APPROVED** [review]
Everything works as expected. Tested on TS1.

**Review by ge65cer — APPROVED** [review]

**Review by Strohgelaender — APPROVED** [review]
Thanks for looking to this!

**Review by timo07k — APPROVED** [review]
Works as expected. Tested on TS1.

**Pull request merged**